nextcord/polls.py

```diff
@@ -0,0 +1,318 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from .enums import PollLayoutType, try_enum
+from .errors import InvalidArgument
+from .partial_emoji import PartialEmoji
+from .user import User
+from .utils import MISSING
+
+if TYPE_CHECKING:
+    from .emoji import Emoji
+    from .message import Message
+    from .state import ConnectionState
+    from .types.polls import (
+        Poll as PollData,
+        PollAnswer as PollAnswerPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollCreateRequest as PollCreateRequestPayload,
+        PollMedia as PollMediaPayload,
+        PollResults as PollResultsPayload,
+    )
+
+    # for reviewers: this typehint type is precedent in the nextcord codebase i.e.
+    # https://github.com/nextcord/nextcord/blob/master/nextcord/message.py#L68
+    EmojiInputType = Union[Emoji, PartialEmoji, str]
+
+
+def resolve_emoji(emoji: EmojiInputType) -> str:
+    if isinstance(emoji, Emoji):
+        return str(emoji.id)
+
+    if isinstance(emoji, PartialEmoji):
+        # if this partial emoji has an ID, then it is a custom emoji. else it's an unicode emoji.
+        return str(emoji.id) if emoji.id else emoji.name
+    if isinstance(emoji, str):
+        return emoji
+
+    raise InvalidArgument(f"Cannot parse `{emoji}` into an emoji to send.")
+
+
+class PollMedia:
+    """
+    A common object that backs both the question and answer.
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of the media.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
+        The emoji of the media.
+
+    """
+
+    __slots__ = (
+        "text",
+        "emoji",
+    )
+
+    def __init__(self, text: str, emoji: Optional[EmojiInputType] = None) -> None:
+        self.text: str = text
+        if emoji:
+            if isinstance(emoji, str):
+                self.emoji: Optional[PartialEmoji | Emoji] = PartialEmoji.from_str(emoji)
+            else:
+                self.emoji: Optional[PartialEmoji | Emoji] = emoji
+        else:
+            self.emoji: Optional[PartialEmoji | Emoji] = emoji  # type: ignore -- pyright is weird?
```

Comment:
This can be simplified to
```py
self.emoji: Optional[PartialEmoji | Emoji] = emoji
if isinstance(emoji, str):
    self.emoji:  Optional[PartialEmoji | Emoji] = PartialEmoji.from_str(emoji)
```